stable_baselines3/common/base_class.py

```diff
@@ -197,7 +197,9 @@ def _wrap_env(env: GymEnv, verbose: int = 0, monitor_wrapper: bool = True) -> Ve
             env = VecTransposeImage(env)
 
         # check if wrapper for dict support is needed when using HER
-        if isinstance(env.observation_space, gym.spaces.dict.Dict):
+        if isinstance(env.observation_space, gym.spaces.dict.Dict) and set(env.observation_space.spaces.keys()) == set(
```

Comment:
note for later: we should probably check that this is a "first-level" dict and throw an error if there is more than one level
The env checker should be updated too

Reply:
I wonder if we need this check at all. Once the DictBuffers are in, would it make the ObsDictWrapper redundant?

Reply:
I meant that as an independent check, your current implementation only supper first-level dict, no?

`ObsDictWrapper` is meant to disappear in the future (when we refactor HER), but let's keep that update separate as this PR is already quite big.

Reply:
Yea, I only support first-level dicts. Ok, I understand :)